### ISSUE TITLE
backward function call

### ISSUE THREAD

**Issue by QiXuanWang** [zackchase/mxnet-the-straight-dope#438]
I noticed in most examples, like those in chapter14, loss.backward() is called within autograd.record() scope. But in some other codes, ( I didn't remember the exact location in this book  but do see it here and other places), loss.backward() is called outside of autograd scope. 
Is there ant specific consideration? I tried and didn't see difference in my own example but I see others reported difference.
